cou/apps/app.py

```diff
@@ -293,6 +379,67 @@ def current_os_release(self) -> OpenStackRelease:
             f"{os_versions}. This is not currently handled."
         )
 
+    @property
+    def is_from_charm_store(self) -> bool:
+        """Check if application comes from charm store.
+
+        :return: True if comes, False otherwise.
+        :rtype: bool
+        """
+        return self.charm_origin == "cs"
+
+    @property
+    def is_subordinate(self) -> bool:
+        """Check if application is subordinate.
+
+        :return: True if subordinate, False otherwise.
+        :rtype: bool
+        """
+        return bool(self.status.subordinate_to)
+
+    @property
+    def is_os_channel_based(self) -> bool:
+        """Check if application is channel based.
+
+        :return: True if does have origin setting, False otherwise.
+        :rtype: bool
+        """
+        if (
+            not self.origin_setting
+            or self.is_versionless
+            or self.is_subordinate
+            or self.charm == "gnocchi"
+        ):
+            return True
+        return False
+
+    @property
+    def is_valid_track(self) -> bool:
+        """Check if the channel track is valid.
+
+        :return: True if valid, False otherwise.
+        :rtype: bool
+        """
+        if self.is_from_charm_store:
+            return True
+        try:
+            track = self.status.charm_channel.split("/", maxsplit=1)[0]
```

Comment:
Why we are using `self.status.charm_channel.split("/", maxsplit=1)[0]` and not `self.channel` as we did above? Also since we are adding so many properties, why not add also track property?

Reply:
`is_valid_track` is used as a validation on `@channel.setter`, so we can't call self.channel because it won't exist. However, we can pass the `charm_channel` and leave as method instead of a property.

I thought about adding a track property. I think it's a good idea